src/components/TextInput/TextInput.tsx

```diff
@@ -218,6 +226,14 @@ export const TextInput: FC<TextInputProps> = ({
                     <LoadingCircle size={LoadingCircleSize.ExtraSmall} />
                 </span>
             )}
+            {validation === Validation.Error && (
```

Comment:
I think this works pretty well now that we have just the error icon in place but we will need soon to have the 3 options, Icon for success and loading. Can we format that to switch the different state easily.
Maybe `validation !== Validation.Default` and render inside the according icon and styling. something like that, WDYT?

Reply:
While I see the valid point of having the code more ready for the next change. I think that change(`validation !== Validation.Default`) should be made when the time comes to implement the loading and success state icons there, otherwise the `render inside the according icon and styling` expression would return nothing in the two other cases. 
Also, the loading state exists atm and it shows the loader on the top right corner. Meaning, the current actual version of only having the error state icon positioned in this place would be less explicit with `validation !== Validation.Default` as the condition.